clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -175,6 +176,13 @@ protected AvroGenericDaVinciClient(
     this.managedClients = managedClients;
     this.icProvider = icProvider;
     this.chunkingAdapter = chunkingAdapter;
+
+    this.recordTransformer = daVinciConfig.isRecordTransformerEnabled() ? daVinciConfig.getRecordTransformer(0) : null;
+
+    if (this.recordTransformer != null) {
+      this.clientConfig.setSpecificValueClass(recordTransformer.getOutputValueClass());
```

Comment:
This might be kind of confusing for users.  Users that specify a specificValue class will have that config silently overridden.  We should either avoid that, or at least log that it's getting overwritten.

I think it might be more prudent to actually compare these.  If they're set to different values throw an error, or, check to see if specificvalueclass is null or not.

Also, can we double check how specificVAlueClass is used?  I seem to recall that this decides whether or not we use a Generic or Specific client? (I might be wrong)

Reply:
What should the desired behavior be if specificValueClass is null, move forward with a generic client?

Reply:
yes